src/bespokelabs/curator/prompter/prompter.py

```diff
@@ -221,4 +221,12 @@ def _get_function_hash(func) -> str:
     if func is None:
         return xxh64("").hexdigest()
 
-    return xxh64(inspect.getsource(func)).hexdigest()
+    return xxh64(_get_function_source(func)).hexdigest()
+
+
+def _get_function_source(func) -> str:
+    """Get the source code of a function."""
+    try:
+        return inspect.getsource(func)
+    except OSError:
```

Comment:
can you document why we're doing this in a comment?

Reply:
yes!